wordpress-mobile/WordPress-iOS
# Add can_view_editor_onboarding property to editor session events

* **Gutenberg:** https://github.com/WordPress/gutenberg/pull/32739
* **Gutenberg Mobile:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/3625
* **WPAndroid:** https://github.com/wordpress-mobile/WordPress-Android/pull/14766

Add `can_view_editor_onboarding` property to editor session events to allow analyzing how capabilities like the editor onboarding impact user behavior. 

To test: 

1. Open Tracks to observe editor session events.
1. Uninstall any preexisting WordPress app from device. 
1. Install WordPress app for this PR. 
1. Launch the Gutenberg block editor. 
1. ℹ️ **Expected:** The "Tap to add content" tooltip is displayed. 
1. Close the Gutenberg block editor. 
1. Launch the Gutenberg block editor. 
1. ℹ️ **Expected:** The "Tap to add content" tooltip is _not_ displayed. 
1. Close the Gutenberg block editor. 
1. ℹ️ **Expected:** Four editor session events are captured: 
    * `wpios_editor_session_start` with `can_view_editor_onboarding: true`.
    * `wpios_editor_session_end` with `can_view_editor_onboarding: true`.
    * `wpios_editor_session_start` with `can_view_editor_onboarding: true`.
    * `wpios_editor_session_end` with `can_view_editor_onboarding: true`. 

## Regression Notes
1. Potential unintended areas of impact
  Modifications to the post editor analytics and publish ability could introduce regressions to publishing or editor-related analytic events. 
2. What I did to test those areas of impact (or what existing automated tests I relied on)
  Manually verified publishing posts worked and related analytic events were triggered for the Gutenberg block editor. 
3. What automated tests I added (or what prevented me from doing so)
  Additional automated tests for this single, experimental property addition are likely not that valuable. 

PR submission checklist:
